File: frontend/src/microservices/advertising/objects/GetAdvertisementDeliverySettingsRequest.ts
// 本文件定义 advertising 模块的 `GetAdvertisementDeliverySettingsRequest`，用于获取投放配置并提供 JSON 编解码。

export type GetAdvertisementDeliverySettingsRequest = {
  placement: string
}

export const getAdvertisementDeliverySettingsRequestFromJson = (json: string): GetAdvertisementDeliverySettingsRequest =>
  JSON.parse(json) as GetAdvertisementDeliverySettingsRequest

export const getAdvertisementDeliverySettingsRequestToJson = (value: GetAdvertisementDeliverySettingsRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/GetAdvertisementDeliverySettingsRequest.scala
// 本文件定义 advertising 模块的 GetAdvertisementDeliverySettingsRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GetAdvertisementDeliverySettingsRequest(
    placement: String
)

object GetAdvertisementDeliverySettingsRequest:
  given sourceEncoder: Encoder[GetAdvertisementDeliverySettingsRequest] = deriveEncoder
  given sourceDecoder: Decoder[GetAdvertisementDeliverySettingsRequest] = deriveDecoder

